lib/Recording/Listener.php

```diff
@@ -54,40 +59,50 @@ public function handle(Event $event): void {
 		};
 	}
 
-	public function handleTranscriptionEvents(AbstractTranscriptionEvent $event): void {
-		if ($event->getAppId() !== Application::APP_ID) {
+	public function handleTranscriptionEvents(AbstractTaskProcessingEvent $event): void {
+		$task = $event->getTask();
+		if ($task->getAppId() !== Application::APP_ID) {
 			return;
 		}
 
-		if ($event instanceof TranscriptionSuccessfulEvent) {
-			$this->successfulTranscript($event->getUserId(), $event->getFile(), $event->getTranscript());
-		} elseif ($event instanceof TranscriptionFailedEvent) {
-			$this->failedTranscript($event->getUserId(), $event->getFile());
-		}
-	}
 
-	protected function successfulTranscript(?string $owner, ?File $fileNode, string $transcript): void {
-		if (!$fileNode instanceof File) {
-			return;
-		}
+		// 'call/transcription/' . $room->getToken()
+		$customId = $task->getCustomId();
+		$isTranscription = $isSummary = false;
+		if (str_starts_with($customId, 'call/transcription/')) {
+			$isTranscription = true;
+			$roomToken = substr($customId, strlen('call/transcription/'));
 
-		if ($owner === null) {
+			$fileId = (int)($task->getInput()['input'] ?? null);
+		} elseif (str_starts_with($customId, 'call/summary/')) {
+			[$roomToken, $fileId] = explode('/', substr($customId, strlen('call/summary/')));
+			$fileId = (int)$fileId;
+			$isSummary = true;
+		} else {
 			return;
 		}
 
-		$this->recordingService->storeTranscript($owner, $fileNode, $transcript);
-	}
-
-	protected function failedTranscript(?string $owner, ?File $fileNode): void {
-		if (!$fileNode instanceof File) {
+		if ($fileId === 0) {
 			return;
 		}
 
-		if ($owner === null) {
+		if ($task->getUserId() === null) {
 			return;
 		}
 
-		$this->recordingService->notifyAboutFailedTranscript($owner, $fileNode);
+		if ($isTranscription) {
```

Comment:
return early on ` !$isTranscription && !$isSummary`

Reply:
it's there?

```php
		} else {
			return;
		}
```
In line 80?